### DISCUSSION TITLE
Opacity property

### DISCUSSION THREAD

**Discussion by bhaloditarun** [QuestPDF/QuestPDF#569] [General]
I'm using image as watermark in all pages.

Do we have any opacity property for image watermark? because text is not readable if image display in background.

Thank you in Advance!

**Comment by MarcinZiabek** [comment]
The easiest way is to convert your image by adding a transparency layer. It can be achieved with SkiaSharp, like so:

```csharp
QuestPDF.Infrastructure.Image LoadImageWithTransparency(string fileName, float transparency)
{
    using var originalImage = SKImage.FromEncodedData(fileName);

    using var surface = SKSurface.Create(originalImage.Width, originalImage.Height, SKColorType.Rgba8888, SKAlphaType.Premul);
    using var canvas = surface.Canvas;                

    using var transparencyPaint = new SKPaint
    {
        ColorFilter = SKColorFilter.CreateBlendMode(SKColors.White.WithAlpha((byte)(transparency * 255)), SKBlendMode.DstIn)
    };
    
    canvas.DrawImage(originalImage, new SKPoint(0, 0), transparencyPaint);

    var encodedImage = surface.Snapshot().Encode(SKEncodedImageFormat.Png, 100).ToArray();
    return Image.FromBinaryData(encodedImage);
}
```

And then used in the foreground page layer, as a watermark:

```csharp
Document.Create(document =>
{
    document.Page(page =>
    {
        page.Content()
            .AlignCenter()
            .Text("Test")
            .FontSize(192)
            .FontColor(Colors.Blue.Medium)
            .Bold();
        
        var image = LoadImageWithTransparency("photo.jpg", 0.75f);
        page.Foreground().Image(image);
    });
})
```

The result is as follows:
<img width="881" alt="Screenshot 2023-05-25 at 00 36 17" src="https://github.com/QuestPDF/QuestPDF/assets/9263853/c15d8309-4f5e-4892-84f2-70f06d826bba">

**Reply by VictorXLR8** [reply]
I want a smaller watermark, to the right of the page. I would also like to control the height of the watermark, so that it does not cover the height of the page.

**Comment by bhaloditarun** [comment]
@MarcinZiabek It's working perfectly.

Very glad for your help and quick reply.

Thank you again :)

**Comment by vmiwouter** [comment]
Is there another way to do this now?
It says SKSurface.Create is Obsolete

**Reply by girlpunk** [reply]
As per the [Skia documentation](https://learn.microsoft.com/en-us/dotnet/api/skiasharp.sksurface.create), `Create(SKImageInfo, SKSurfaceProperties)` should be used instead of `Create(int, int, SKColorType, SKAlphaType, SKSurfaceProps)`
